controller/src/functions/localization.ts

```diff
@@ -0,0 +1,154 @@
+  const {scripturalRenderingMetadata, rendering, content, language, gitRepo} =
+    dbSchema;
+  //@ this is the raw sql version of what the orm below is doing
+  // const query = sql.raw(`SELECT book_name, book_slug, ietf_code, id
+  // FROM (SELECT book_name, book_slug, l.ietf_code, c.id,
+  // ROW_NUMBER() OVER (PARTITION BY l.ietf_code, book_slug ORDER BY book_slug) AS rn
+  //     FROM scriptural_rendering_metadata AS srm
+  //     JOIN rendering AS r ON r.id = srm.rendering_id
+  //     JOIN content AS c ON r.content_id = c.id
+  //     JOIN language AS l ON l.ietf_code = c.language_id
+  //     JOIN git_repo AS gr ON c.git_id = gr.id
+  //     WHERE gr.username ILIKE '%wa-catalog%'
+  //     AND c.domain = 'scripture'
+  //     AND book_slug IS NOT NULL
+  // ) AS subquery
+  // WHERE rn = 1
+  // ORDER BY ietf_code, book_slug;`);
+
+  const subquery = db
+    .select({
+      book_name: scripturalRenderingMetadata.bookName,
+      book_slug: scripturalRenderingMetadata.bookSlug,
+      ietf_code: language.ietfCode,
+      id: content.id,
+      rn: sql
+        .raw(
+          "ROW_NUMBER() OVER (PARTITION BY ietf_code, book_slug ORDER BY book_slug)"
+        )
+        .as("rn"),
+    })
+    .from(scripturalRenderingMetadata)
+    .leftJoin(
+      rendering,
+      eq(rendering.id, scripturalRenderingMetadata.renderingId)
+    )
+    .leftJoin(content, eq(content.id, rendering.contentId))
+    .leftJoin(language, eq(language.ietfCode, content.languageId))
+    .leftJoin(gitRepo, eq(content.gitId, gitRepo.id))
+    .where(
+      and(
+        ilike(gitRepo.username, "%wa-catalog%"),
+        eq(content.domain, "scripture"),
+        isNotNull(scripturalRenderingMetadata.bookSlug)
+      )
+    )
+    .as("sq");
+  const result = await db
+    .select({
+      book_name: subquery.book_name,
+      book_slug: subquery.book_slug,
+      ietf_code: subquery.ietf_code,
+      id: subquery.id,
+    })
+    .from(subquery)
+    .where(eq(subquery.rn, 1))
+    .orderBy(subquery.ietf_code, subquery.book_slug);
+
+  // rowList should be book_name, book_slug, ietf_code, id (content id that is);
+  // zod validate this;
+
+  const parsed = z
+    .array(
+      z.object({
+        book_name: z.string(),
+        book_slug: z.string(),
+        ietf_code: z.string(),
+        id: z.string(),
+      })
+    )
+    .parse(result);
+  const payload: insertLocalizationType[] = parsed.map((row) => {
+    return {
+      ietfCode: row.ietf_code,
+      key: row.book_slug.toLowerCase(),
+      value: row.book_name,
+    };
+  });
+  const inserted = await polymorphicInsert({
+    tableKey: "localization",
+    content: payload,
+    onConflictDoUpdateArgs: {
+      target: [table.ietfCode, table.key],
+      set: onConflictSetAllFieldsToSqlExcluded(table, ["ietfCode", "key"]),
+    },
+  });
+  return inserted;
+}
+
+app.timer("manageLocalizationTable", {
+  schedule: "0 0 0 * * *",
+  handler: populateLocalization,
+  useMonitor: false,
+});
```

Comment:
We can ruin this not on a cron, but for now outside of setting up something more complicated such as doing event driven inserts based on the metadata table, this is a really straightforward way, and this day surely isn't gonna be that time critical I'd image.